packages/toolpad-core/src/SignInPage/SignInPage.tsx

```diff
@@ -0,0 +1,324 @@
+import { BrandingContext } from '../AppProvider';
+import { useNotifications } from '../useNotifications';
+
+const IconProviderMap = new Map<string, React.ReactNode>([
+  ['github', <GitHubIcon key="github" />],
+  ['credentials', <PasswordIcon key="credentials" />],
+  ['google', <GoogleIcon key="google" />],
+  ['facebook', <FacebookIcon key="facebook" />],
+]);
+
+export interface AuthProvider {
+  /**
+   * The unique identifier of the authentication provider.
+   * @default ''
+   * @example 'google'
+   * @example 'github'
+   */
+  id: string;
+  /**
+   * The name of the authentication provider.
+   * @default ''
+   * @example 'Google'
+   * @example 'GitHub'
+   */
+  name: string;
+}
+
+export interface SignInPageProps {
+  /**
+   * The list of authentication providers to display.
+   * @default []
+   */
+  providers?: AuthProvider[];
+  /**
+   * Callback fired when a user signs in.
+   * @param {AuthProvider} provider The authentication provider.
+   * @param {FormData} formData The form data if the provider id is 'credentials'.\
+   * @param {string} callbackUrl The URL to redirect to after signing in.
+   * @returns {void} | {string}
+   * @default undefined
+   */
+  signIn?: (provider: AuthProvider, formData?: any, callbackUrl?: string) => void | Promise<string>;
+  /**
+   * Props to pass to the constituent components in the credentials form
+   * @default {}
+   * @example { email: { autoFocus: false } }
+   * @example { password: { variant: 'outlined' } }
+   * @example { email: { autoFocus: false }, password: { variant: 'outlined' } }
+   */
+  componentProps?: {
+    email?: TextFieldProps;
+    password?: TextFieldProps;
+    button?: LoadingButtonProps;
+  };
+}
+
+/**
+ *
+ * Demos:
+ *
+ * - [Sign In Page](https://mui.com/toolpad/core/react-sign-in-page/)
+ *
+ * API:
+ *
+ * - [SignInPage API](https://mui.com/toolpad/core/api/sign-in-page)
+ */
+function SignInPage(props: SignInPageProps) {
+  const { providers, signIn, componentProps } = props;
+  const branding = React.useContext(BrandingContext);
+  const credentialsProvider = providers?.find((provider) => provider.id === 'credentials');
+  const [{ loading, providerId }, setFormStatus] = React.useState<{
+    loading: boolean;
+    providerId: string;
+  }>({
+    providerId: '',
+    loading: false,
+  });
+
+  const callbackUrl =
+    typeof window !== 'undefined'
+      ? new URLSearchParams(window.location.search).get('callbackUrl') ?? '/'
+      : '/';
+
+  const notifications = useNotifications();
+
+  return (
+    <Container component="main" maxWidth="xs">
+      <Box
+        sx={{
+          marginTop: 8,
+          display: 'flex',
+          flexDirection: 'column',
+          alignItems: 'center',
+        }}
+      >
+        {branding?.logo ?? (
+          <Avatar sx={{ m: 1, bgcolor: 'primary.main' }}>
+            <LockOutlinedIcon />
+          </Avatar>
+        )}
+
+        <Typography variant="h5" color="textPrimary" gutterBottom>
```

Comment:
It looks great, I think I'd just try to make the title a bit bolder and bigger by default to see how it looks. Maybe a bit more spacing under the logo too but that's just nitpicking and not sure if needed :D

Reply:
Agreed with the spacing; for the text, I tried different heading levels and `h5` seems the most suitable:
<img width="849" alt="h5" src="https://github.com/user-attachments/assets/a77515bf-2ed5-4d1b-a33d-793d61f977f1">
<img width="851" alt="h4" src="https://github.com/user-attachments/assets/3dc15c3f-8ca0-4c65-9c37-071152c788a1">

Reply:
Looks good!